iridauploader/core/upload.py

```diff
@@ -153,7 +183,13 @@ def batch_upload_single_entry(batch_directory, force_upload=False, upload_mode=N
     error_list = []
     for directory_status in upload_list:
         logging.info("Starting upload for {}".format(directory_status.directory))
-        result = _validate_and_upload(directory_status, upload_mode)
+        result = _validate_and_upload(directory_status, upload_mode, False)
+        if result.exit_code == exit_return.EXIT_CODE_ERROR:
+            error_list.append(directory_status)
+
+    for directory_status in upload_as_partial_list:
+        logging.info("Continuing upload from partial for {}".format(directory_status.directory))
+        result = _validate_and_upload(directory_status, upload_mode, True)
         if result.exit_code == exit_return.EXIT_CODE_ERROR:
             error_list.append(directory_status)
```

Comment:
Rather than keeping the 2 lists here can we just keep one `upload_list` and somehow have the status stored with it?  These 2 loops are identical save for the logging statement and the `True/False` in the `_validate_and_upload`.  It just looks a bit funny.

Reply:
Good catch, I made a simple dict for the status + partial boolean, simplifying the loops quite a bit.
The comment for partial/upload has been moved to `_validate_and_upload`